### ISSUE TITLE
Gap between screens.

### ISSUE THREAD

**Issue by alvarlagerlof** [britzl/monarch#24]
I get a big gap between screens with the slide animation. See video. 

Also, is it possible to make custom transitions?
[2018-06-16 23-39-31.zip](https://github.com/britzl/monarch/files/2108526/2018-06-16.23-39-31.zip)

**Comment by britzl** [comment]
That is because the transition speed isn't linear. The main menu screen that is about to be hidden will accelerate while the level select screen will decelerate. This causes a gap when the difference in speed between the two screens is the largest.

Yes, you can create custom transitions:

https://github.com/britzl/monarch#predefined-transitions

You can create a linear transition or a transition where both hide and show uses the same easing function. I assume you have something like this in the gui_script of both of your screens:

```
local transitions = require "monarch.transitions.gui"
local monarch = require "monarch.monarch"

function init(self)
	-- create transitions for the node 'root'
	-- the node will slide in/out from left and right with
	-- a specific easing, duration and delay
	self.transition = transitions.create(gui.get_node("root"))
		.show_in(transitions.slide_in_right, gui.EASING_OUTQUAD, 0.6, 0)
		.show_out(transitions.slide_out_left, gui.EASING_INQUAD, 0.6, 0)
		.back_in(transitions.slide_in_left, gui.EASING_OUTQUAD, 0.6, 0)
		.back_out(transitions.slide_out_right, gui.EASING_INQUAD, 0.6, 0)
end
```

The solution is to change from gui.EASING_OUTQUAD and gui.EASING_INQUAD to matching easing functions. Perhaps gui.EASING_LINEAR on both, or gui.EASING_INQUAD on both.

**Comment by alvarlagerlof** [comment]
Ah, thank you. Now it's clearer. I also have the problem of Animations looking different when the next screen is first loaded versus when it's loaded and I open it again. Is there somehow a way to fix this. 

If I optimize for the second time i get a gap the first time.

**Comment by britzl** [comment]
Ah, the preload() function doesn't detect if trying to preload an already loaded screen. Fixed this in Monarch 2.9.1: https://github.com/britzl/monarch/releases/tag/2.9.1

**Comment by alvarlagerlof** [comment]
Ah, great!
